File: cases/taint/java/expression-negative/ExpressionNegative.java
package dataflowbench.taint;

final class ExpressionNegative {
    static int dfb_source() { // DFB-SOURCE: expression-negative-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: expression-negative-sink

    static void run() {
        int value = dfb_source();
        int computed = (value * 3) + 7; // DFB-WITNESS: expression-negative-computed
        dfb_sink(7);
    }
}

File: cases/taint/go/expression-negative/expression_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: expression-negative-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: expression-negative-sink

func run() {
	value := dfb_source()
	computed := (value * 3) + 7 // DFB-WITNESS: expression-negative-computed
	_ = computed
	dfb_sink(7)
}
